### PR TITLE
Anko - Change Share screen from xml to anko layout

### PR DESCRIPTION
**Pull request by at-datbui**
#### Related issues (Required)
- Task #110

#### Description (Required)
- [x]  Change xml to anko layout

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 1]
```diff
@@ -1,21 +1,46 @@
...
+
+    companion object {
+        const val ID_RL_BOTTOM_CARD = 1
+        const val ID_BTN_CLOSE = 2
+        const val ID_TV_TITLE = 3
+        const val ID_TV_DESCRIPTION = 4
+        const val ID_BTN_SHARING = 5
+    }
+
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 2]
```diff
@@ -34,9 +59,112 @@ class BottomButtonCard @JvmOverloads constructor(context: Context, attrs: Attrib
...
+                    alignParentBottom()
+                }
+                btnClose = rippleView {
+                    id = ID_BTN_CLOSE
+                    bottomPadding = dimen(R.dimen.padding_medium)
+                    leftPadding = dimen(R.dimen.padding_medium)
+                    rightPadding = dimen(R.dimen.padding_medium)
+                    imageView {
+                        imageResource = R.drawable.ic_navigation_close
+                    }.lparams(dip(15), dip(15))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 3]
```diff
@@ -34,9 +59,112 @@ class BottomButtonCard @JvmOverloads constructor(context: Context, attrs: Attrib
...
+                        centerInParent()
+                    }
+
+                    imgLoader = imageView {
+                        imageResource = R.drawable.ic_live_location_loading
+                        visibility = View.GONE
+                    }.lparams(wrapContent, wrapContent) {
+                        centerInParent()
+                    }
+                }.lparams(matchParent, dip(50)) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 4]
```diff
@@ -34,9 +59,112 @@ class BottomButtonCard @JvmOverloads constructor(context: Context, attrs: Attrib
...
+                    tvCopyLink = textView(R.string.bottom_button_card_text_copy_link) {
+                        textColor = ContextCompat.getColor(context, R.color.black)
+                        backgroundResource = R.drawable.custom_bg_button_copy
+                        padding = dimen(R.dimen.padding_base)
+                        setAllCaps(false)
+                    }.lparams(wrapContent, wrapContent) {
+                        alignParentRight()
+                        margin = dimen(R.dimen.margin_low)
+                    }
+                }.lparams(matchParent, dip(50)) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Extension method to hide keyboard.
+ */
+fun View.hideKeyboard(context: Context) {
+    val imm: InputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+    imm.hideSoftInputFromWindow(this.windowToken, 0)
+}
+
+internal fun ViewManager.mapView(init: MapView.() -> Unit): MapView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    return ankoView({ MapView(it) }, 0, init)
+}
+
+internal fun ViewManager.rippleView(init: com.hypertrack.lib.internal.consumer.view.RippleView.() -> Unit):
+        com.hypertrack.lib.internal.consumer.view.RippleView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    return ankoView({ com.hypertrack.lib.internal.consumer.view.RippleView(it, null) }, 0, init)
+}
+
+internal fun ViewManager.wrappingViewPager(init: com.hypertrack.lib.internal.consumer.view.WrappingViewPager.() -> Unit):
+        com.hypertrack.lib.internal.consumer.view.WrappingViewPager {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    return ankoView({ com.hypertrack.lib.internal.consumer.view.WrappingViewPager(it) }, 0, init)
+}
+
+internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Created by haingoq on 12/10/2017.
+ */
+
+/**
+ * Extension method to hide keyboard.
+ */
+fun View.hideKeyboard(context: Context) {
+    val imm: InputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    return ankoView({ com.hypertrack.lib.internal.consumer.view.RippleView(it, null) }, 0, init)
+}
+
+internal fun ViewManager.wrappingViewPager(init: com.hypertrack.lib.internal.consumer.view.WrappingViewPager.() -> Unit):
+        com.hypertrack.lib.internal.consumer.view.WrappingViewPager {
+    return ankoView({ com.hypertrack.lib.internal.consumer.view.WrappingViewPager(it) }, 0, init)
+}
+
+internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
+    return ankoView({ BottomButtonCard(it) }, 0, init)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Checks whether files end with a line separator.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    imm.hideSoftInputFromWindow(this.windowToken, 0)
+}
+
+internal fun ViewManager.mapView(init: MapView.() -> Unit): MapView {
+    return ankoView({ MapView(it) }, 0, init)
+}
+
+internal fun ViewManager.rippleView(init: com.hypertrack.lib.internal.consumer.view.RippleView.() -> Unit):
+        com.hypertrack.lib.internal.consumer.view.RippleView {
+    return ankoView({ com.hypertrack.lib.internal.consumer.view.RippleView(it, null) }, 0, init)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+internal fun ViewManager.wrappingViewPager(init: com.hypertrack.lib.internal.consumer.view.WrappingViewPager.() -> Unit):
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+        com.hypertrack.lib.internal.consumer.view.WrappingViewPager {
+    return ankoView({ com.hypertrack.lib.internal.consumer.view.WrappingViewPager(it) }, 0, init)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 8 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 14]
```diff
@@ -1,21 +1,48 @@
...
+        const val ID_RL_BOTTOM_CARD = 1
+        const val ID_BTN_CLOSE = 2
+        const val ID_TV_TITLE = 3
+        const val ID_TV_DESCRIPTION = 4
+        const val ID_BTN_SHARING = 5
+        const val LAYOUT_HEIGHT = 50
+        const val IMAGE_SIZE = 15
+    }
+
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+    imm.hideSoftInputFromWindow(this.windowToken, 0)
+}
+
+internal fun ViewManager.mapView(init: MapView.() -> Unit): MapView {
+    return ankoView({ MapView(it) }, 0, init)
+}
+
+internal fun ViewManager.rippleView(
+        init: com.hypertrack.lib.internal.consumer.view.RippleView.() -> Unit):
+        com.hypertrack.lib.internal.consumer.view.RippleView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+    return ankoView(
+            { com.hypertrack.lib.internal.consumer.view.RippleView(it, null) }, 0, init)
+}
+
+internal fun ViewManager.wrappingViewPager(
+        init: com.hypertrack.lib.internal.consumer.view.WrappingViewPager.() -> Unit):
+        com.hypertrack.lib.internal.consumer.view.WrappingViewPager {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+    return ankoView(
+            { com.hypertrack.lib.internal.consumer.view.WrappingViewPager(it) }, 0, init)
+}
+
+internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 18]
```diff
@@ -34,9 +61,112 @@ class BottomButtonCard @JvmOverloads constructor(context: Context, attrs: Attrib
...
+                    backgroundResource = R.drawable.custom_bg_button_share
+
+                    tvStartShare = textView(R.string.bottom_button_card_view_text_start_share) {
+                        textSize = px2dip(dimen(R.dimen.text_large))
+                        gravity = Gravity.CENTER
+                    }.lparams(wrapContent, wrapContent) {
+                        centerInParent()
+                    }
+
+                    imgLoader = imageView {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: Please change this line to `imgLoader = imageView(R.drawable.ic_live_location_loading) {` and remove line 123

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 19]
```diff
@@ -34,9 +61,112 @@ class BottomButtonCard @JvmOverloads constructor(context: Context, attrs: Attrib
...
+                        centerVertically()
+                        visibility = View.GONE
+                        leftMargin = dimen(R.dimen.margin_xxhigh)
+                    }
+
+                    tvCopyLink = textView(R.string.bottom_button_card_text_copy_link) {
+                        textColor = ContextCompat.getColor(context, R.color.black)
+                        backgroundResource = R.drawable.custom_bg_button_copy
+                        padding = dimen(R.dimen.padding_base)
+                        setAllCaps(false)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 20]
```diff
@@ -0,0 +1,129 @@
...
+import vn.asiantech.way.extension.rippleView
+import vn.asiantech.way.extension.wrappingViewPager
+
+/**
+ * Copyright © 2017 Asian Tech Co., Ltd.
+ * Created by datbuit. on 27/11/2017.
+ */
+class ShareActivityUI : AnkoComponent<ShareActivity> {
+    companion object {
+        const val ID_MAP = 1
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: Please correct id

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+        const val ID_BTN_BACK = 2
+        const val ID_TV_TITLE = 3
+        const val MARGIN_VALUE_VERY_SMALL = 5
+        const val MARGIN_VALUE_SMALL = 10
+        const val MARGIN_VALUE_MEDIUM = 15
+        const val MARGIN_VALUE_LARGE = 20
+        const val MARGIN_VALUE_VERY_LARGE = 50
+    }
+
+    internal lateinit var mapView: MapView
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: Please gather all variable that same type

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+    internal lateinit var imgCurrentLocation: ImageView
+    internal lateinit var imgChooseMarker: ImageView
+
+    override fun createView(ui: AnkoContext<ShareActivity>) = with(ui) {
+        relativeLayout {
+            lparams(matchParent, matchParent)
+            frameLayout {
+                lparams(matchParent, matchParent)
+                relativeLayout {
+                    mapView = mapView {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: change `mapview` to `framelayout`

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+                        id = ID_MAP
+                    }.lparams(matchParent, matchParent)
+
+                    rippleView {
+                        id = ID_BTN_BACK
+                        imageView {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: Please remove line 52

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+                            textSize = px2dip(dimen(R.dimen.text_medium))
+                            leftPadding = dip(MARGIN_VALUE_LARGE)
+                            rightPadding = dip(MARGIN_VALUE_VERY_LARGE)
+                            bottomPadding = dip(MARGIN_VALUE_SMALL)
+                        }.lparams(matchParent, wrapContent) {
+                            below(ID_TV_TITLE)
+                        }
+
+                        imgEdit = imageView {
+                            imageResource = R.drawable.ic_edit_location
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 25]
```diff
@@ -0,0 +1,129 @@
...
+                    wrappingViewPager {
+                    }.lparams(matchParent, wrapContent) {
+                        alignParentBottom()
+                        bottomMargin = dimen(R.dimen.margin_high)
+                    }
+
+                    imgCurrentLocation = imageView {
+                        scaleType = ImageView.ScaleType.FIT_XY
+                        visibility = View.GONE
+                        imageResource = R.drawable.ic_ht_reset_button
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 26]
```diff
@@ -0,0 +1,129 @@
...
+                            topPadding = dip(MARGIN_VALUE_MEDIUM)
+                            rightPadding = dip(MARGIN_VALUE_LARGE)
+                        }.lparams(wrapContent, wrapContent) {
+                            alignParentRight()
+                        }
+                    }.lparams(matchParent, wrapContent) {
+                        rightOf(ID_BTN_BACK)
+                    }
+
+                    wrappingViewPager {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: remove this view

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 27]
```diff
@@ -0,0 +1,129 @@
...
+                    }.lparams(dimen(R.dimen.set_bounds_icon_size), dimen(R.dimen.set_bounds_icon_size)) {
+                        alignParentEnd()
+                        alignParentRight()
+                        alignWithParent = true
+                        margin = dip(MARGIN_VALUE_SMALL)
+                    }
+
+                    imgChooseMarker = imageView {
+                        visibility = View.GONE
+                        imageResource = R.drawable.select_expected_place
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 28]
```diff
@@ -0,0 +1,129 @@
...
+                        padding = dip(MARGIN_VALUE_SMALL)
+                    }.lparams(wrapContent, wrapContent) {
+                        centerInParent()
+                    }
+                }.lparams(matchParent, matchParent) {
+                    backgroundColor = ContextCompat.getColor(context, R.color.colorWhite)
+                }
+
+                imgChooseMarker = imageView {
+                    imageResource = R.drawable.select_expected_place
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt [hunk 29]
```diff
@@ -0,0 +1,129 @@
...
+                }.lparams(wrapContent, wrapContent) {
+                    bottomMargin = dip(MARGIN_VALUE_SMALL)
+                    gravity = Gravity.CENTER
+                }
+            }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivityUI.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 30]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Created by haingoq on 12/10/2017.
+ * Extension method to hide keyboard.
+ */
+fun View.hideKeyboard(context: Context) {
+    val imm: InputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+    imm.hideSoftInputFromWindow(this.windowToken, 0)
+}
+
+internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 31]
```diff
@@ -0,0 +1,27 @@
...
+    return ankoView(
+            { RippleView(it, null) }, 0, init)
+}
+
+internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 32]
```diff
@@ -0,0 +1,27 @@
...
+import com.hypertrack.lib.internal.consumer.view.RippleView
+import org.jetbrains.anko.custom.ankoView
+import vn.asiantech.way.ui.custom.BottomButtonCard
+
+/**
+ * Created by haingoq on 12/10/2017.
+ * Extension method to hide keyboard.
+ */
+fun View.hideKeyboard(context: Context) {
+    val imm: InputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 33]
```diff
@@ -0,0 +1,27 @@
...
+    imm.hideSoftInputFromWindow(this.windowToken, 0)
+}
+
+internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
+    return ankoView(
+            { RippleView(it, null) }, 0, init)
+}
+
+internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
+    return ankoView({ BottomButtonCard(it) }, 0, init)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 34]
```diff
@@ -0,0 +1,360 @@
...
+
+        initiateView()
+
+    }
+
+    /**
+     * Enum define for item type
+     */
+    enum class ActionType {
+        START_TRACKING,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 35]
```diff
@@ -0,0 +1,360 @@
...
+        CONFIRM_LOCATION,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 36]
```diff
@@ -0,0 +1,360 @@
...
+        SHARE_TRACKING_URL,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 37]
```diff
@@ -0,0 +1,360 @@
...
+        SHARE_BACK_LOCATION
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 38]
```diff
@@ -0,0 +1,360 @@
...
+        tvStartShare.text = actionText
+    }
+
+    internal fun showBottomCardLayout() {
+        hideProgress()
+        btnSharing.visibility = View.VISIBLE
+        AnimationUtils.expand(this, AnimationUtils.DURATION_DEFAULT_VALUE_ANIMATION)
+    }
+
+    fun hideBottomCardLayout() {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public functions require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 39]
```diff
@@ -0,0 +1,360 @@
...
+import android.widget.RelativeLayout
+import android.widget.TextView
+import com.hypertrack.lib.internal.common.util.HTTextUtils
+import com.hypertrack.lib.internal.consumer.utils.AnimationUtils
+import com.hypertrack.lib.internal.consumer.view.RippleView
+import org.jetbrains.anko.*
+import vn.asiantech.way.R
+import vn.asiantech.way.extension.rippleView
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 30 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 31 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 32 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 33 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 40]
```diff
@@ -0,0 +1,360 @@
...
+
+        initiateView()
+
+    }
+
+    /*
+     * Enum define for item type
+     */
+    enum class ActionType {
+        START_TRACKING,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 41]
```diff
@@ -0,0 +1,360 @@
...
+        CONFIRM_LOCATION,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 42]
```diff
@@ -0,0 +1,360 @@
...
+        SHARE_TRACKING_URL,
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 43]
```diff
@@ -0,0 +1,360 @@
...
+        SHARE_BACK_LOCATION
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt (hunk 39 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING: Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 44]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Created by haingoq on 12/10/2017.
+ * Extension method to hide keyboard.
+ */
+fun View.hideKeyboard(context: Context) {
+    val imm: InputMethodManager? = context.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
+    imm?.hideSoftInputFromWindow(this.windowToken, 0)
+}
+
+internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 45]
```diff
@@ -0,0 +1,27 @@
...
+    return ankoView(
+            { RippleView(it, null) }, 0, init)
+}
+
+internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 46]
```diff
@@ -0,0 +1,27 @@
...
+    return ankoView({ BottomButtonCard(it) }, 0, init)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 47]
```diff
@@ -0,0 +1,360 @@
...
+    }
+
+    private fun initiateView() {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: Please init listener when create view

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 48]
```diff
@@ -0,0 +1,360 @@
...
+//            }
+//        }
+//        imgBtnCall.setOnClickListener {
+//            if (onBottomCardListener != null) {
+//                onBottomCardListener?.onCallButtonClick()
+//            }
+//        }
+    }
+
+    internal fun setTitleText(title: String) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: Please change all inner fun return `BottomCard...`

File: app/src/main/java/vn/asiantech/way/ui/share/ShareActivity.kt [hunk 49]
```diff
@@ -0,0 +1,21 @@
...
+import org.jetbrains.anko.setContentView
+import vn.asiantech.way.ui.base.BaseActivity
+
+/**
+ * Copyright © 2017 Asian Tech Co., Ltd.
+ * Created by datbuit. on 27/11/2017.
+ */
+class ShareActivity : BaseActivity() {
+
+    private lateinit var ui: ShareActivityUI
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/share/ShareActivity.kt]: Change variable name to `shareActivityUI`

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finished review this PR, please check it again.
thank you!

**Pull request merged**